### PR TITLE
ENT-3514 Refactor out duplication between blackboard and canvas Integrated channel codebase

### PR DESCRIPTION
**Pull request by binodpant**
History is that when we did blackboard we reused much of the code from canvas so things started to look very duplicatish, we already cleaned out some, this is just remaining vestiges now turned out not much to be done

Duplication being addressed:

* `client#_create_session` is identical in the two so can be easily refactored to use the existing function `refresh_session_if_expired`, blackboard already does that
* exporter for learner data looks almost identical between the two however refactor to DRY does not make a lot of sense right now so simply moved lines of code around so they both read similar and make easier if we want to DRY in the future

**Merge checklist:**
- [ ] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `base.in` if needed in production but edx-platform doesn't install it
    - `test-master.in` if edx-platform pins it, with a matching version
    - `make upgrade && make requirements` have been run to regenerate requirements
- [ ] `make static` has been run to update webpack bundling if any static content was updated
- [ ] `./manage.py makemigrations` has been run
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
    - This should be run from either a venv with all the lms/edx-enterprise requirements installed or if you checked out edx-enterprise into the src directory used by lms, you can run this command through an lms shell.
        - It would be `./manage.py lms makemigrations` in the shell.
- [x] [Version](https://github.com/edx/edx-enterprise/blob/master/enterprise/__init__.py) bumped
- [x] [Changelog](https://github.com/edx/edx-enterprise/blob/master/CHANGELOG.rst) record added
- [ ] Translations updated (see docs/internationalization.rst but also this isn't blocking for merge atm)

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/edx/edx-enterprise/releases) released
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
- [ ] After versioned build finishes in [GitHub Actions](https://github.com/edx/edx-enterprise/actions), verify version has been pushed to [PyPI](https://pypi.org/project/edx-enterprise/)
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
- [ ] PR created in [edx-platform](https://github.com/edx/edx-platform) to upgrade dependencies (including edx-enterprise)
    - This **must** be done after the version is visible in PyPi as `make upgrade` in edx-platform will look for the latest version in PyPi.
    - Note: the edx-enterprise constraint in edx-platform **must** also be bumped to the latest version in PyPi.

### CODE REVIEW

File: integrated_channels/canvas/client.py [hunk 1]
```diff
@@ -505,12 +481,20 @@ def _handle_get_user_canvas_course(self, canvas_user_id, learner_data_course_id)
...
-    def _get_oauth_access_token(self, client_id, client_secret):
-        """Uses the client id, secret and refresh token to request the user's auth token from Canvas.
+    def _create_session(self):
+        """
+        Instantiate a new session object for use in connecting with Canvas. Each enterprise customer
+        connecting to Canvas should have a single client session.
+        Will only create a new session if token expiry has been reached
+        """
+        self.session, self.expires_at = refresh_session_if_expired(
+            self._get_oauth_access_token,
```
Comment (alex-sheehan-edx) [inline integrated_channels/canvas/client.py:491]: Oh I see, you're able to make this a util for both Canvas and blackboard by passing in their individual token retrieval methods 👍 . I'm very in favor of this!

File: integrated_channels/canvas/exporters/learner_data.py [hunk 2]
```diff
@@ -32,32 +32,34 @@ def get_learner_data_records(
...
                 'No learner data was sent for user [%s] because a Canvas user ID could not be found.',
-                enterprise_enrollment.enterprise_customer_user.username
+                enterprise_customer_user.username
             )
             return None
         percent_grade = kwargs.get('grade_percent', None)
         completed_timestamp = completed_date.strftime("%F") if isinstance(completed_date, datetime) else None
 
+        course_catalog_client = get_course_catalog_api_service_client(
+            site=enterprise_customer_user.enterprise_customer.site
```
Comment (alex-sheehan-edx) [inline integrated_channels/canvas/exporters/learner_data.py:46]: 👍

**Review by alex-sheehan-edx — APPROVED** [review]
nice cleanup! 👍  lgtm

File: integrated_channels/canvas/client.py (hunk 1 above)
Comment (binodpant) [reply integrated_channels/canvas/client.py:491]: yeah I had forgotten we had this method then reviewing side by side just saw they were identical as long as I can pasa f unctoin to get oauth_access_token so works out!

**Pull request merged**